### ISSUE TITLE
Automatic discovery of the length of a chain of successive join transactions

### ISSUE THREAD

**Issue by raedah** [raedahgroup/dcrchainanalysis#6]
We should be able to recognize a series of successive joins on the the blockchain. Starting at an intiially given tx, we would trace back its history or origin to see how many prior join transactions it is the child of.

**Comment by samparsky** [comment]
@raedah Is this where we would have to specify the depth of tracing an output?
Also can you explain what you mean by`join transactions`?

**Comment by raedah** [comment]
Yes, one possible initial implementation would be to simply specify a depth of how far into the history you want to scan. It would likely be sane to have a max depth to scan in any case. A join transaction can be defined as a transaction with multiple inputs and multiple outputs, where the outputs are at least 2 of an identical amount which would make it impossible to know the source input with certainty.
